### PR TITLE
UCT/UD_ACCEL: fix calc of max short am threshold

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe pls take a look

### CODE REVIEW

**Comment by yosefe** [comment]
@MattBBaker can you pls post ORNL failure?

File: src/uct/ib/mlx5/ib_mlx5.h [hunk 1]
```diff
@@ -68,13 +68,18 @@ struct mlx5_grh_av {
...
+      (_av_size) + \
+      sizeof(struct mlx5_wqe_inl_data_seg)))
+
 #define UCT_IB_MLX5_AM_MAX_HDR(_av_size) \
-        UCT_IB_MLX5_MAX_BB * MLX5_SEND_WQE_BB - \
-        (sizeof(struct mlx5_wqe_ctrl_seg) + \
-        (_av_size) + \
-        sizeof(struct mlx5_wqe_inl_data_seg) + \
-        UCT_IB_MLX5_AM_ZCOPY_MAX_IOV * sizeof(struct mlx5_wqe_data_seg))
+    (UCT_IB_MLX5_AM_MAX_SHORT(_av_size) - \
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.h]: let's rename the macro to UCT_IB_MLX5_AM_MAX_ZCOPY_HDR?

**Comment by MattBBaker** [comment]
looks like a registration failure.
```
[----------] 10 tests from test_rcache, where TypeParam = 
[ RUN      ] test_rcache.merge_merge_unwritable/mt_6
[       OK ] test_rcache.merge_merge_unwritable/mt_6 (6 ms)
[ RUN      ] test_rcache.merge_invalid_prot/mt_6
[       OK ] test_rcache.merge_invalid_prot/mt_6 (6 ms)
[ RUN      ] test_rcache.release_inuse/mt_6
[       OK ] test_rcache.release_inuse/mt_6 (12 ms)
[ RUN      ] test_rcache.merge_with_unwritable/mt_6
[       OK ] test_rcache.merge_with_unwritable/mt_6 (6 ms)
[ RUN      ] test_rcache.get_unmapped/mt_6
[1511800996.795025] [pod9:12512:0]         rcache.c:479  UCX  WARN  failed to register region 0x386ec90 [0x7f3dd8dbc010..0x7f3dd8ebc010]: Input/output error
/var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/ucs/test_rcache.cc:55: Failure
Error: Input/output error
terminate called after throwing an instance of 'ucs::test_abort_exception'
  what():  std::exception
make: *** [test] Aborted
make: Leaving directory `/var/lib/jenkins/jobs/UCX/workspace/build-tests/1511800547/test/gtest'
```

**Comment by yosefe** [comment]
BTW we can't assume smaller AV size, in case one of the endpoints would have to send to a L3 destination..

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5260/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5265/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5274/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5278/ for details (Mellanox internal link).

**Pull request merged**